students/nDruP/lesson09/mailroom.py

```diff
@@ -0,0 +1,223 @@
+
+
+def main_menu(user_prompt = None):
+    """
+    Prompt user to send a Thank You, Create a report, create letters, or quit.
+    """
+    valid_prompts = {"1": create_thank_u,
+                     "2": create_donor_report,
+                     "3": write_letters_to_all,
+                     "4": exit}
+    options = list(valid_prompts.keys())
+    print(divider + "We're a Pyramid Scheme & So Are You! E-Mailroom"
+          + divider)
+    while user_prompt not in valid_prompts:
+        options_str = ("{}" + (", {}") * (len(options)-1)).format(*options)
+        print(f"Please choose from the following options ({options_str}):")
+        print("1. Send a Thank you")
+        print("2. Create Donor Report")
+        print("3. Send letters to everyone")
+        print("4. Quit")
+        user_prompt = input(">")
+        print(divider)
+    return valid_prompts.get(user_prompt)
+
+
+def user_input(some_str = ""):
+    """
+    Display exit reminder and prompt user for input.
+    """
+    while not some_str:
+        print("Return to the main menu by entering 'exit'")
+        some_str = input(">")
+    return check_not_exit(some_str) * some_str
+
+
+def check_not_exit(check_str):
+    """
+    Check whether or not given string is "exit"
+    """
+    return check_str.lower() != "exit"
+
+
+def input_dir(write_dir=""):
+    """
+    Prompt user for a valid path to directory. 
+    """
+    while True:
+        print("Enter an existing directory to write to.")
+        print("Input 'cwd' to use Current Working Directory")
+        write_dir = user_input()
+        if write_dir.lower() == "cwd":
+            write_dir = os.getcwd()
+        if os.path.isdir(write_dir) or not write_dir:
+            break
+    return write_dir
+
+
+def save_to_dir(save_name, save_text):
+    """
+    Prompt user for choice to save previously printed text to drive.
+    """
+    save_txt = " "
+    while save_txt not in ['y', 'n', '']:
+        print("Would you like to save previously displayed text"
+              " to a file [y/n]?")
+        save_txt = user_input()
+    if save_txt == 'y':
+        write_dir = input_dir()
+        if write_dir:
+            return write_txt_to_dir(save_name, save_text, write_dir)
+    return ""
+
+
+def write_txt_to_dir(f_name, content, wrt_dir=os.getcwd()):
+    """
+    Write a personalized thank you letter for all the donors.
+    Letters will be written to letter_dir.
+    """
+    curdate = (datetime.datetime.now()).strftime("%Y_%m_%d")
+    file_name = f_name.replace(' ', '_') + "_" + curdate + ".txt"
+    file_path = os.path.join(wrt_dir, file_name)
+    with open(file_path, 'w+') as text:
+        text.write(content)
+    return "Wrote to " + file_path
+
+
+def conv_str(conv_str, conv_type=int):
+    """
+    Convert string to given conv_type.
+    If it's unable to convert, return original string.
+    """
+    try:
+        conv_yes = conv_type(conv_str)
+        return conv_yes
+    except:
```

Comment:
It's almost always a good idea to hone in on a specific exception like `ValueError`